### DISCUSSION TITLE
Clone depth of 1, pull all branches

### DISCUSSION THREAD

**Discussion by elliot-wasem** [gabrie30/ghorg#277] [Ideas]
I was wondering if it's feasible to clone at a depth of 1, and pull down all non-deleted branches from remote.

My reason is that I'm trying to get around gitlab's inability to search for files by name across all repos in my organization (short of self-hosting and using elasticsearch). So, I want to be able to clone down all repos (thank you `ghorg`), and search.

However, cloning all my org's repos takes forever. And therefore comes my desire for `--depth=1` or similar.

I am planning on using `git worktree` to connect branches, then using `fzf` to actually do searches.

Anyway, does this sound feasible as a `ghorg` feature? Or should I just implement that myself?

**Comment by gabrie30** [comment]
Sorry @elliot-wasem I didn't see this until today. Yes I think this would be feasible. Probably a flag like `--clone-depth=n` would work. You might also want to omit any blobs if you are searching ghorg can already do that using `--git-filter=blob:none`
